src/Plugin/Field/FieldFormatter/StrawberryPagedFormatter.php

```diff
@@ -705,6 +703,65 @@ public function processElementforMetadatadisplays(
     return $element;
   }
 
+  /**
+   * Generates render element for a manifest URL.
+   *
+   * @param int $delta
+   * @param array $jsondata
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *
+   * @return array
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function processElementforManifestURL($delta = 0, array $jsondata, FieldItemInterface $item) {
+    $element = [];
+    $entity = NULL;
+    $nodeuuid = $item->getEntity()->uuid();
+    $max_width = $this->getSetting('max_width');
+    $max_height = $this->getSetting('max_height');
+
+    if ($this->getSetting('manifesturl_source')) {
+	      $manifest_url_key = $this->getSetting('manifesturl_source');
+      	if ($jsondata[$manifest_url_key]) {
+		      $manifest_url = $jsondata[$manifest_url_key];
```

Comment:
Question: do you think we need to allow many URLS in that key? Is there a need for that?

Reply:
Since we don't have a validation on the value, should we try to make sure its an URL? I like this function https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21UrlHelper.php/function/UrlHelper%3A%3AisValid/8.6.x to test that and the use statement is already there =) https://github.com/esmero/format_strawberryfield/blob/master/src/Plugin/Field/FieldFormatter/StrawberryPagedFormatter.php#L18

Reply:
Answer: at this time I don't imagine need with more than 1 manifest per book.
Great function, applied to manifest URL check if is valid, code updated.

Reply:
I forgot I supposed an absolute URL for manifest so I modified variable description adding "absolute".